source/fuzz/fuzzer_pass_wrap_regions_in_selections.cpp

```diff
@@ -0,0 +1,159 @@
+        merge_block_candidates[GetFuzzerContext()->RandomIndex(
+            merge_block_candidates)]);
+    if (!merge_block_candidate) {
+      continue;
+    }
+
+    if (!TransformationWrapRegionInSelection::IsApplicableToBlockRange(
+            GetIRContext(), header_block_candidate->id(),
+            merge_block_candidate->id())) {
+      continue;
+    }
+
+    // This boolean constant will be used as a condition for the
+    // OpBranchConditional instruction. We mark it as irrelevant to be able to
+    // replace it with a more interesting value later.
+    auto branch_condition = GetFuzzerContext()->ChooseEven();
+    FindOrCreateBoolConstant(branch_condition, true);
+
+    ApplyTransformation(TransformationWrapRegionInSelection(
+        header_block_candidate->id(), merge_block_candidate->id(),
+        branch_condition));
+  }
+}
+
+opt::BasicBlock*
+FuzzerPassWrapRegionsInSelections::MaybeGetHeaderBlockCandidate(
+    opt::BasicBlock* header_block_candidate) {
+  // Try to create a preheader if |header_block_candidate| is a loop header.
+  if (header_block_candidate->IsLoopHeader()) {
+    // GetOrCreateSimpleLoopPreheader only supports reachable blocks.
+    return GetIRContext()->cfg()->preds(header_block_candidate->id()).size() ==
+                   1
+               ? nullptr
+               : GetOrCreateSimpleLoopPreheader(header_block_candidate->id());
+  }
+
+  // Try to split |header_block_candidate| if it's already a header block.
+  if (header_block_candidate->GetMergeInst()) {
+    auto* split_before_inst = &*header_block_candidate->begin();
+    while (split_before_inst->opcode() == SpvOpVariable ||
+           split_before_inst->opcode() == SpvOpPhi) {
+      split_before_inst = split_before_inst->NextNode();
+    }
+
+    ApplyTransformation(TransformationSplitBlock(
+        MakeInstructionDescriptor(GetIRContext(), split_before_inst),
+        GetFuzzerContext()->GetFreshId()));
+  }
+
+  return header_block_candidate;
+}
+
+opt::BasicBlock* FuzzerPassWrapRegionsInSelections::MaybeGetMergeBlockCandidate(
+    opt::BasicBlock* merge_block_candidate) {
+  // If |merge_block_candidate| is a merge block of some construct, try to split
+  // it and return a newly created block.
+  if (GetIRContext()->GetStructuredCFGAnalysis()->IsMergeBlock(
+          merge_block_candidate->id())) {
+    if (merge_block_candidate->IsLoopHeader()) {
+      // We can't split a merge block if it's also a loop header.
+      return nullptr;
+    }
+
+    auto* split_before_inst = &*merge_block_candidate->begin();
+    while (split_before_inst->opcode() == SpvOpPhi) {
+      split_before_inst = split_before_inst->NextNode();
+    }
+
+    auto new_block_id = GetFuzzerContext()->GetFreshId();
+    ApplyTransformation(TransformationSplitBlock(
+        MakeInstructionDescriptor(GetIRContext(), split_before_inst),
+        new_block_id));
+    return GetIRContext()->cfg()->block(new_block_id);
```

Comment:
#3790 will introduce:

```
  // Returns the second block in the pair obtained by splitting |block_id| just
  // after the last OpPhi or OpVariable instruction in it. Assumes that the
  // block is not a loop header.
  opt::BasicBlock* SplitBlockAfterOpPhiOrOpVariable(uint32_t block_id);
```

Could that be used here (once it's merged)?

Reply:
Yes, I'll fix it when it's merged.

Reply:
Fixed in f4d8d40.